aten/src/ATen/native/Resize.cpp

```diff
@@ -35,18 +35,21 @@ static bool is_functorch_wrapped_tensor(const Tensor& tensor) {
   return !(key_set & kFunctorchWrappedTensors).empty();
 }
 
-bool resize_output(const Tensor& output, IntArrayRef shape) {
+bool resize_output(const Tensor& output, IntArrayRef shape, c10::optional<IntArrayRef> strides) {
   if (resize_output_check(output, shape)) {
-    // avoid a redispatch for cpu and cuda.
-    // TODO: when resize_cuda_ is re-written to be unified with resize_,
-    // we can provide the same benefit for cuda.
-    //
     // TODO(#61485): functorch wrapped tensors should not go through the
     // fast path. This is a hack, longer term solutions are in the issue
     if (output.is_cpu() && !is_functorch_wrapped_tensor(output)) {
-      at::native::resize_(output, shape);
+      at::native::resize_template<&resize_impl_tryreuse_<&maybe_resize_storage_cpu>>(
```

Comment:
Is there any reason not to do something like
```
      if (strides.has_value()) {
        at::native::resize_(output, shape, strides.value());
      } else {
        at::native::resize_(output, shape);
      }
```

It's a little daunting to stare at that template when trying to understand the fastpath logic (but all we really care about is skipping the dispatch hop)

Reply:
I thought that by using the templates directly, I wouldn't need the branch. But, now that you mentioned it, I think it's way better to leave a branch there.